services/API-service/src/api/event/event.controller.ts

```diff
@@ -0,0 +1,25 @@
+import { EventPcodeDto } from './dto/event-pcode.dto';
+import { EventService } from './event.service';
+import { Body, Controller, Post, UseGuards } from '@nestjs/common';
+import { ApiBearerAuth, ApiOperation, ApiTags } from '@nestjs/swagger';
+import { RolesGuard } from '../../roles.guard';
+
+@ApiBearerAuth()
+@UseGuards(RolesGuard)
+@ApiTags('event')
+@Controller('event')
+export class EventController {
+  private readonly eventService: EventService;
+
+  public constructor(eventService: EventService) {
+    this.eventService = eventService;
+  }
+
+  @ApiOperation({ summary: 'Check EAP actions as done' })
+  @Post('close-pcode')
+  public async closeEventPcode(
+    @Body() eventPcodeDto: EventPcodeDto,
+  ): Promise<void> {
+    return await this.eventService.closeEventPcode(eventPcodeDto);
```

Comment:
Should there be a response message? We could send the updated object so the interface does not have to do a hard reload.

Reply:
For now I just removed the area that you close from the fthisilteredAreas object from the interface. 
` this.filteredAreas = this.filteredAreas.filter(
      (item) => item.id !== eventPlaceCodeId,
    );`
So it does not show up in the interface anymore. Do you think that is good enough or do you think we should redo the api call to get open events from the backend?

Reply:
I think we should redo the API call.